src/Application/HttpModels/DonationCreate.php

```diff
@@ -6,6 +6,7 @@
 
 /**
  * Request-only payload for setting up new donations.
+ * @psalm-suppress MissingConstructor Seems we can't use an enum w/o default without hitting this.
```

Comment:
True we can't use an enum w/o default without hitting this, Psalm wants to make sure we never hit any Uninitialized property errors, which I think is a good thing. I'm not sure why you don't want to let it default to card as it was.

Reply:
I think we probably shouldn't suppress the MissingConstructor error - I think making sure we initialise all required properties in constructors is a really good thing Psalm can do for us. Can we either set the default to card to match what it was doing before, or if you want to improve it add a constructor, e.g. as below and pass the arguments to it from tests? 

That will also make the symfony seralizer ensure those three arguments are always provided:

https://symfony.com/doc/current/components/serializer.html#handling-constructor-arguments

```
    public function __construct(
        string $donationAmount,
        PaymentMethodType $paymentMethodType,
        string $projectId,
        string $psp
    ) {
        $this->donationAmount = $donationAmount;
        $this->paymentMethodType = $paymentMethodType;
        $this->projectId = $projectId;
        $this->psp = $psp;
    }
    ```